modules/core/src/main/java/org/apache/ignite/internal/processors/rest/handlers/cache/GridCacheCommandHandler.java

```diff
@@ -363,15 +392,16 @@ private static Object appendOrPrepend(Object origVal, Object appendVal,
 
         GridRestCacheRequest req0 = (GridRestCacheRequest)req;
 
-        final String cacheName = req0.cacheName() == null ? DFLT_CACHE_NAME : req0.cacheName();
-
         final Object key = req0.key();
 
         final Set<GridClientCacheFlag> cacheFlags = GridClientCacheFlag.parseCacheFlags(req0.cacheFlags());
 
         try {
             GridRestCommand cmd = req0.command();
 
+            if (req0.cacheName() == null && CACHE_NAME_REQUIRED_REQUESTS.contains(cmd))
```

Comment:
For some requests absence of cache name is correct scenario, so we should check them separately. I was following the logic of GridCacheCommandHandler with its KEY_REQUIRED_REQUESTS. But you are right, it will be more safe to create array with requests that should be processed differently. In such case the default behavior will be to check cache name.